packages/medusa/src/repositories/sales-channel.ts

```diff
@@ -30,4 +30,18 @@ export class SalesChannelRepository extends Repository<SalesChannel> {
 
     return await qb.getManyAndCount()
   }
+
+  async removeProducts(
+    salesChannelId: string,
+    productIds: string[]
+  ): Promise<DeleteResult> {
+    return await this.createQueryBuilder()
+      .delete()
+      .from("product_sales_channel")
+      .where({
+        sales_channel_id: salesChannelId,
+        product_id: In(productIds),
+      })
+      .execute()
```

Comment:
question: is this idempotent? i.e. if one of `productIds` doesn't exist will it ignore or fail?

Reply:
it is idempotent and should ignore if no tuple match the criteria, just like a raw sql query 👍 I have duplicated the query in the integration tests to enforce the indempotency, so if it changes somehow, we would know.